examples/graphql-subscriptions/graphql_subscriptions.bal

```diff
@@ -16,7 +13,29 @@ service /graphql on new graphql:Listener(9090) {
     // A resource method with the `subscribe` accessor represents a field in the root
     // `Subscription` operation. It must always return a stream. Since the stream is of type
     // `string`, the resulting field in the generated GraphQL schema will be of type `String!`.
-    resource function subscribe names() returns stream<string> {
-        return self.names.toStream();
+    resource function subscribe names() returns stream<string, error?> {
+        // Create a NameGenerator class object
+        NameGenerator nameGenerator = new (self.names);
+        // Create a stream using the NameGenerator class object
+        stream<string, error?> names = new (nameGenerator);
+        return names;
+    }
+}
+
+// Defines a StreamGenerator class that can be used to create a stream of strings. This will pick a random name from
+// the list of names and return it with a delay to demonstrate a stream of values.
+class NameGenerator {
+    private final string[] names;
+
+    isolated function init(string[] names) {
+        self.names = names;
+    }
+
+    // Function that picks a random name from the list and returns it.
```

Comment:
```suggestion
    // The `next` method picks a random name from the list and returns it.
```

Reply:
This suggestion because `next` is something specifically required in a stream implementor.